plugins/bulk-import/src/components/AddRepositories/AddRepositoriesForm.tsx

```diff
@@ -1,133 +1,112 @@
-// import RadioGroup from '@mui/material/RadioGroup';
-// import Tooltip from '@mui/material/Tooltip';
-// import Typography from '@mui/material/Typography';
-// import { useFormikContext } from 'formik';
-// import { AddRepositoriesFormValues } from '../../types';
-import { AddRepositoriesFormFooter } from './AddRepositoriesFormFooter';
-import { AddRepositoriesTable } from './AddRepositoriesTable';
+import { DrawerContextProvider } from '@janus-idp/shared-react';
 
-const useStyles = makeStyles(theme => ({
-  body: {
-    marginBottom: '50px',
-    padding: '24px',
-  },
-  approvalTool: {
-    display: 'flex',
-    flexDirection: 'row',
-    justifyContent: 'left',
-    alignItems: 'center',
-    paddingTop: '24px',
-    paddingBottom: '24px',
-    paddingLeft: '16px',
-    backgroundColor: theme.palette.background.paper,
-    borderBottomStyle: 'groove',
-    border: theme.palette.divider,
-  },
+import { bulkImportApiRef } from '../../api/BulkImportBackendClient';
+import {
+  AddRepositoriesFormValues,
+  ApprovalTool,
+  CreateImportJobRepository,
+  ImportJobResponse,
+  RepositorySelection,
+} from '../../types';
+import {
+  getJobErrors,
+  prepareDataForSubmission,
+} from '../../utils/repository-utils';
+import { AddRepositories } from './AddRepositories';
 
-  approvalToolTooltip: {
-    paddingTop: '4px',
-    paddingRight: '24px',
-    paddingLeft: '5px',
-  },
-}));
+export const AddRepositoriesForm = () => {
+  const bulkImportApi = useApi(bulkImportApiRef);
+  const navigate = useNavigate();
+  const [generalSubmitError, setGeneralSubmitError] = React.useState<{
+    message: string;
+    title: string;
+  } | null>(null);
+  const initialValues: AddRepositoriesFormValues = {
+    repositoryType: RepositorySelection.Repository,
+    repositories: {},
+    excludedRepositories: {},
+    approvalTool: ApprovalTool.Git,
+  };
 
-export const AddRepositoriesForm = ({
-  error,
-}: {
-  error: { message: string; title: string } | null;
-}) => {
-  const styles = useStyles();
-  const { openDrawer, setOpenDrawer, drawerData } = useDrawer();
-  const { setFieldValue, values } =
-    useFormikContext<AddRepositoriesFormValues>();
+  const createImportJobs = async (importOptions: {
+    importJobs: CreateImportJobRepository[];
+    dryRun?: boolean;
+  }) =>
+    await bulkImportApi.createImportJobs(
+      importOptions.importJobs,
+      importOptions.dryRun,
+    );
 
-  const closeDrawer = () => {
-    setOpenDrawer(false);
-  };
+  const mutationCreate = useMutation(createImportJobs, {
+    onSuccess: (data: ImportJobResponse[] | Response) => {
+      return data;
+    },
+    onError: (error: Error) => {
+      setGeneralSubmitError({
+        message:
+          error?.message ||
+          `${get(error, 'error.message')}\n${get(error, 'error.response.url')}` ||
+          'Failed to create pull request',
+        title: error?.name || get(error, 'error.name') || 'Error occured',
+      });
+      return error;
+    },
+  });
```

Comment:
I think it's easier if you handle this later by using the mutationCreate.error

```suggestion
  const mutationCreate = useMutation(createImportJobs);
```